### DISCUSSION TITLE
currentPage

### DISCUSSION THREAD

**Discussion by InnuendoPi** [sstaub/NextionX2#5] [Q&A] [answered]
First: thx a lot for this lib! 
how can I get a reliable information about the current (active) page? Send component IDs on touch or release events are not a reliable way while moving pages up and down. always ends up i a wrong pageID. (my env: softserial@9600, Wemos D1 mini). 
Innu

**Comment by sstaub** [comment, marked as answer]
In the moment there is no function available. I made an issue for this but I don't know when I find some time to solve it.
The solution is not very complicated.

**Reply by InnuendoPi** [reply]
thank you for opening an issue. there is no need to hurry! 
Edit: A quick'n'dirty solution [https://github.com/InnuendoPi/NextionX2](url)
in Nextion editor add _printh 23 xx 00_ in preinit event to all pages, replace xx with the page id (in hex eg 01, 02)

**Reply by JOKaija** [reply]
3 year old messages, but there is a couple other ways to do this too.

You can use sys values and make a little routine to read those sys0...sysxxx values. Then you can uses ie. sys0 for page information.
Set it as sys0=pagenumber on every page preinit event.
You can find example code by google how to read those sys values.
OR
Create dummy number value to page 0. Set it as global and change text and background colors to hide it. Update its value on every page preinit. You can read it any time, even you are not at page 0 with normal commands. Ie. 
----
NextionComponent pagenum(nextion, 0, 1);
int currentpage=pagenum.attributeValue("val");

And in HMI page preinits as:

page0.n0.val=0
page0.n0.val=1
etc....
